src/main/docs/guide/kafkaStreams/kafkaStreamExceptions.adoc

```diff
@@ -0,0 +1,26 @@
+Since version 2.8.0, Kafka allows you to https://kafka.apache.org/28/javadoc/org/apache/kafka/streams/errors/StreamsUncaughtExceptionHandler.html[handle uncaught exceptions] that may be thrown from your streams. This handler must return the action that must be taken, depending on the thrown exception.
+
+There are three possible https://kafka.apache.org/28/javadoc/org/apache/kafka/streams/errors/StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.html[responses]: `REPLACE_THREAD`, `SHUTDOWN_CLIENT`, or `SHUTDOWN_APPLICATION`.
+
+TIP: You can find more details about this mechanism https://developer.confluent.io/tutorials/error-handling/confluent.html[here].
+
+To implement your own handler, you can listen to the application event api:io.micronaut.configuration.kafka.streams.event.BeforeKafkaStreamStart[] and configure the streams with your own business logic:
+
+[source,java]
+----
+include::{testskafkastreams}/uncaught/MyStreamsUncaughtExceptionHandler.java[tags=imports, indent=0]
+
+include::{testskafkastreams}/uncaught/MyStreamsUncaughtExceptionHandler.java[tags=clazz, indent=0]
+----
+
+However, if you just want to take the same action every time, you can set the application property `kafka.streams.[STREAM-NAME].uncaught-exception-handler` to a valid action, such as `REPLACE_THREAD`.
```

Comment:
Should we perhaps reverse the order of these approaches / examples? I think it's good to show the simpler option first.

Reply:
Yeah, that's a good idea. Thanks!